edge-function/src/helpers/body.rs

```diff
@@ -0,0 +1,143 @@
+use crate::bindings::wasi::http::types::IncomingBody;
+use anyhow::Result;
+use bytes::Bytes;
+
+pub trait FromBody: Sized {
+    fn from_body(body: IncomingBody) -> Result<Self>;
+}
+
+impl FromBody for () {
+    fn from_body(_: IncomingBody) -> Result<Self> {
+        Ok(())
+    }
+}
+
+impl FromBody for IncomingBody {
+    fn from_body(body: IncomingBody) -> Result<Self> {
+        Ok(body)
+    }
+}
+
+impl FromBody for Bytes {
+    fn from_body(body: IncomingBody) -> Result<Self> {
+        body.read()
+    }
+}
+
+pub trait IntoBody: Sized {
+    fn into_body(self) -> Result<Bytes>;
+    fn handle_error(err: anyhow::Error) -> http::Response<Bytes>;
+
+    #[allow(unused_variables)]
+    fn extend_response_parts(&self, parts: &mut http::response::Parts) {}
+}
+
+impl IntoBody for Bytes {
+    fn into_body(self) -> Result<Bytes> {
+        Ok(self)
+    }
+
+    fn handle_error(_: anyhow::Error) -> http::Response<Bytes> {
+        http::Response::builder()
+            .status(http::StatusCode::INTERNAL_SERVER_ERROR)
+            .body(Bytes::new())
+            .unwrap()
+    }
+}
+
+impl IntoBody for () {
+    fn into_body(self) -> Result<Bytes> {
+        Ok(Bytes::new())
+    }
+
+    fn handle_error(err: anyhow::Error) -> http::Response<Bytes> {
+        Bytes::handle_error(err)
+    }
+}
+
+impl IntoBody for String {
+    fn into_body(self) -> Result<Bytes> {
+        Ok(Bytes::from(self))
+    }
+
+    fn handle_error(err: anyhow::Error) -> http::Response<Bytes> {
+        Bytes::handle_error(err)
+    }
+}
+
+// Data types
```

Comment:
I've also added this to support optional values (that's what I need for the Stripe component)

```suggestion
// Data types

impl<T: IntoBody> IntoBody for Option<T> {
    fn into_body(self) -> Result<Bytes> {
        match self {
            Some(value) => {
                T::into_body(value)                    
            }
            None => Ok(Bytes::new()),
        }
    }

    fn handle_error(err: anyhow::Error) -> http::Response<Bytes> {
        T::handle_error(err)
    }

    fn extend_response_parts(&self, parts: &mut http::response::Parts) {
        match self {
            Some(value) => {
                T::extend_response_parts(value, parts);
            }
            None => {}, // do nothing?
        }
        
    }
}

impl<T: FromBody> FromBody for Option<T> {
    fn from_body(body: IncomingBody) -> Result<Self> {
        match T::from_body(body) {
            Ok(value) => {
                Ok(Some(value))
            },
            Err(_) => return Ok(None),
        }
    }
}
```

Feel free to simplify this ^_^